lib/notesDisplay.ml

```diff
@@ -735,55 +747,84 @@ let print_misc_notes conf base =
-          |> Output.print_sstring conf)
-        (build_path_hierarchy d);
-
-    let current_depth =
-      if d = "" then 0
-      else List.length (String.split_on_char NotesLinks.char_dir_sep d) + 1
+  let one_folder r =
+    let simple_d =
+      if d = "" then r else d ^ String.make 1 NotesLinks.char_dir_sep ^ r
     in
-
-    List.iter
-      (function
-        | _, Some f ->
-            let txt =
-              let n, s = read_notes base f in
-              let t = try List.assoc "TITLE" n with Not_found -> "" in
-              if t <> "" then (Util.escape_html t :> string)
-              else if s = "" then ""
-              else
-                "<em>"
-                ^ (begin_text_without_html_tags 50 s |> Util.escape_html
-                    :> string)
-                ^ "</em>"
-            in
-            let n, _ = read_notes base f in
-            let n_type = try List.assoc "TYPE" n with Not_found -> "" in
-            let simple_f = format_simple_filename d f in
-            format_file_entry conf current_depth d simple_f n_type txt
-            |> Output.print_sstring conf
-        | r, None ->
-            format_folder_entry conf current_depth r
-              (if d = "" then r
-              else d ^ String.make 1 NotesLinks.char_dir_sep ^ r)
-              false false
-            |> Output.print_sstring conf)
-      db;
-    Output.print_sstring conf "</div>");
-
+    format_folder_entry conf current_depth r simple_d false false
+    |> Output.print_sstring conf
+  in
+  let one_file f =
+    let envn, s = read_notes base f in
+    let txt =
+      let t = try List.assoc "TITLE" envn with Not_found -> "" in
+      if t <> "" then (Util.escape_html t :> string)
+      else if s = "" then ""
+      else
+        "<em>"
+        ^ (begin_text_without_html_tags 50 s |> Util.escape_html :> string)
+        ^ "</em>"
+    in
+    let n_type = try List.assoc "TYPE" envn with Not_found -> "" in
+    let simple_f = format_simple_filename d f in
+    format_file_entry conf current_depth d simple_f n_type txt
+    |> Output.print_sstring conf
+  in
+  let dirs_in_db, files_in_db =
+    if db <> [] then (
+      Output.print_sstring conf {|<div class="px-1">|};
+      if d <> "" then (
+        format_folder_entry conf 0 ".." "" false true
+        |> Output.print_sstring conf;
+        path_hierarchy d);
+      let dirs_in_db, files_in_db =
+        List.fold_left
+          (fun (dirs_in_db, files_in_db) entry ->
+            match entry with
+            | _, Some f ->
+                one_file f;
+                (dirs_in_db, f :: files_in_db)
+            | r, None ->
+                one_folder r;
+                (r :: dirs_in_db, files_in_db))
+          ([], []) db
+      in
+      Output.print_sstring conf "</div>";
+      (dirs_in_db, files_in_db))
+    else ([], [])
+  in
+  let d =
+    String.split_on_char NotesLinks.char_dir_sep d
+    |> String.concat Filename.dir_sep
```

Comment:
I think that we should write a function to convert system paths into note links paths and vice versa.